ai-compile/vscode-extension/src/code-lens/codeLensProvider.ts

```diff
@@ -0,0 +1,42 @@
+import * as vscode from 'vscode';
+import * as t from "../types";
+import { storageManager } from '../extension';
+
+
+export class CodelensProvider implements vscode.CodeLensProvider {
+
+    private codeLenses: vscode.CodeLens[] = [];
+    private regex: RegExp;
+    private _onDidChangeCodeLenses: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+    public readonly onDidChangeCodeLenses: vscode.Event<void> = this._onDidChangeCodeLenses.event;
+
+    constructor() {
+        // non-empty source lines
+        this.regex = /(.+)/g;
+        vscode.workspace.onDidChangeConfiguration( (_) => {
```

Comment:
I think we use a slightly different whitespace style. I'll add `prettier` and we should just do the code formatting through that tool.